crates/router/src/routes/app.rs

```diff
@@ -161,6 +163,12 @@ impl SessionStateInfo for SessionState {
         self.store.add_request_id(request_id.to_string());
         self.request_id.replace(request_id);
     }
+    fn get_header_map(&self) -> Option<HeaderMap> {
+        self.api_client.get_header_map()
+    }
+    fn add_header_map(&mut self, header_map: HeaderMap) {
+        self.api_client.add_header_map(header_map);
+    }
```

Comment:
Add `request_headers` as a field on `SessionState`, and remove these methods from the `SessionStateInfo` trait. 

Implement these as methods instead on `SessionState` alone, something like:

```rust
impl SessionState {
    pub fn get_request_headers(&self) -> &http::HeaderMap {
        &self.request_headers
    }

    pub fn set_request_headers(&mut self, headers: http::HeaderMap) {
        // ...
    }
}
```